pkg/splunk/enterprise/util.go

```diff
@@ -287,7 +287,7 @@ func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.Me
 		}
 	}
 
-	if localAppsConf != yamlConfHeader {
+	if localAppsConf != yamlConfHeader && localAppsConf != yamlConfLocalHeader {
 		mapAppListing["app-list-local.yaml"] = localAppsConf
 	}
 
```

Comment:
`The app-list configmap only adds app that have changes since the last app install. If there are no apps to add, the file is not written to the configmap.`

@jryb Why isn't this a concern for clustered case?

Reply:
The issue is that for CRs where there can be a `cluster` and `local` scope (CMs and Deployer), for local apps we uses the new `apps_location_local` variable in ansible.  So we need to not write the configmap if that file is blank, but now a blank file either looks like:
```
splunk:
  apps_location:
```
or 
```
splunk:
  apps_location_local:
```
So we need to check for both.